lib/vmpooler/dns/gcp.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+require 'googleauth'
+require 'google/cloud/dns'
+require 'vmpooler/dns/base'
+
+module Vmpooler
+  class PoolManager
+    class Dns
+      # This class represent a DNS plugin to CRUD resources in google clouddns.
+      class Gcp < Vmpooler::PoolManager::Dns::Base
+        # The connection_pool method is normally used only for testing
+        attr_reader :connection_pool
+
+        def initialize(config, logger, metrics, redis_connection_pool, name, options)
+          super(config, logger, metrics, redis_connection_pool, name, options)
+
+          task_limit = global_config[:config].nil? || global_config[:config]['task_limit'].nil? ? 10 : global_config[:config]['task_limit'].to_i
+
+          default_connpool_size = [provided_pools.count, task_limit, 2].max
+          # connpool_size = provider_config['connection_pool_size'].nil? ? default_connpool_size : provider_config['connection_pool_size'].to_i
+          connpool_timeout = 60
+          # logger.log('d', "[#{name}] ConnPool - Creating a connection pool of size #{connpool_size} with timeout #{connpool_timeout}")
```

Comment:
What's with the commented out bits?

Reply:
Leftover copy/paste, I think if it needs customized outside of the computed size then that can be added as an additional feature.